cfy_manager/components/restservice/restservice.py

```diff
@@ -355,20 +410,74 @@ def _upload_cloudify_license():
 
     def install(self):
         logger.notice('Installing Rest Service...')
-        self._install()
+        yum_install(config[RESTSERVICE][SOURCES]['restservice_source_url'])
+        yum_install(config[RESTSERVICE][SOURCES]['agents_source_url'])
+
+        self._chown_resources_dir()
+
+        set_logrotate(RESTSERVICE)
+
+        if DATABASE_SERVICE not in config[SERVICES_TO_INSTALL]:
+            yum_install(config[RESTSERVICE][SOURCES]['haproxy_rpm_url'])
         logger.notice('Rest Service successfully installed')
 
     def configure(self):
         logger.notice('Configuring Rest Service...')
-        self._configure()
+        if (
+            DATABASE_SERVICE not in config[SERVICES_TO_INSTALL]
+            and config[POSTGRESQL_CLIENT]['host'] in (
+                '127.0.0.1', 'localhost',
+            )
+            and config[POSTGRESQL_SERVER]['cluster']['nodes']
+        ):
+            self._configure_db_proxy()
+
+        try:
+            if config[CLEAN_DB]:
+                self._set_admin_password()
+                self._generate_flask_security_config()
+            else:
+                self._validate_admin_password_and_security_config()
+            self._make_paths()
+            self._configure_restservice()
+            self._enter_sanity_mode()
+            self._configure_db()
+            systemd.configure(RESTSERVICE)
+            systemd.restart(RESTSERVICE)
+            if config[CLUSTER_JOIN]:
+                logger.info('Extra node in cluster, will verify rest-service '
+                            'after clustering configured')
+            else:
+                self._verify_restservice_alive()
+        finally:
+            self._exit_sanity_mode()
+
         self._upload_cloudify_license()
         logger.notice('Rest Service successfully configured')
 
     def remove(self):
         logger.notice('Removing Restservice...')
         systemd.remove(RESTSERVICE, service_file=False)
         remove_logrotate(RESTSERVICE)
-        self._remove_files()
+
+        yum_remove('cloudify-rest-service')
+        yum_remove('cloudify-agents')
+
+        common.remove('/opt/manager')
+
+        if (
+            DATABASE_SERVICE not in config[SERVICES_TO_INSTALL]
```

Comment:
this 5-line condition is used twice now, so it should be factored out to a method so that we can give it a descriptive name

Reply:
My usual approach is to allow it twice and then factor it out if it's needed a third time, but I'll take a quick look in case it might be clearer factored out.

Reply:
Factored out into a utils function.